sycl/plugins/unified_runtime/ur/adapters/cuda/memory.cpp

```diff
@@ -485,3 +482,90 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemBufferPartition(
   *phMem = MemObj.release();
   return UR_RESULT_SUCCESS;
 }
+
+ur_result_t
+ur_buffer_::allocateMemObjOnDeviceIfNeeded(ur_device_handle_t hDevice) {
+  UR_ASSERT(hDevice, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+
+  ScopedDevice Active(hDevice);
+  ur_lock_guard LockGuard(MemoryAllocationMutex);
+
+  ur_result_t Result = UR_RESULT_SUCCESS;
+  CUdeviceptr &DevPtr = getNativePtr(hDevice);
+
+  // Allocation has already been made
+  if (DevPtr != ur_buffer_::native_type{0}) {
+    return UR_RESULT_SUCCESS;
+  }
+
+  if (MemAllocMode == ur_buffer_::AllocMode::AllocHostPtr) {
+    // Host allocation has already been made
+    Result = UR_CHECK_ERROR(cuMemHostGetDevicePointer(&DevPtr, HostPtr, 0));
```

Comment:
No need for `Result` (here and below), as `UR_CHECK_ERROR` either throws or aborts.

Reply:
Nice catch thanks